src/empirical.jl

```diff
@@ -51,18 +51,43 @@ struct EmpiricalVariogram
     sums   = zeros(nlags)
     counts = zeros(Int, nlags)
 
+    # decide whether to construct a spatial index
+    # coordinates need to be AbstractFloat for NearestNeighbors to work
+    # https://github.com/KristofferC/NearestNeighbors.jl/issues/13
+    isfloat = eltype(coordinates(sdata)) <: AbstractFloat
```

Comment:
You can use `coordtype(sdata)` to get the type of the coordinates without building the array.

Reply:
Good tip, done in 0009ea940c63a829c0238850f89467d7e5bf8557